device/esp_tinyusb/descriptors_control.c

```diff
@@ -178,8 +178,8 @@ esp_err_t tinyusb_set_descriptors(const tinyusb_config_t *config)
 
     // Select FullSpeed configuration descriptor
     if (config->configuration_descriptor == NULL) {
-        // Default configuration descriptor is provided only for CDC, MSC and NCM classes
-#if (CFG_TUD_HID > 0 || CFG_TUD_MIDI > 0 || CFG_TUD_CUSTOM_CLASS > 0 || CFG_TUD_ECM_RNDIS > 0 || CFG_TUD_DFU > 0 || CFG_TUD_DFU_RUNTIME > 0 || CFG_TUD_BTH > 0)
+        // Default configuration descriptor must be provided for the following classes
```

Comment:
Does it mean if using the following classes in esp_tinyusb, the user has no chance to self-define the configuration descriptor?

Reply:
It means the opposite. If the user implements HID. MIDI, DFU, BTH... classes he must provide a self-defined configuration descriptor.

For CDC, Vendor Specific, MSC and NCM classes we provide a default descriptor that the user can override with self-defined descriptor